btk/measure.py

```diff
@@ -268,11 +273,24 @@ def __next__(self):
         measure_results = {}
         for i, f in enumerate(self.measure_functions):
             measure_dic = {}
-            for key in ["catalog", "deblended_images", "segmentation"]:
+            for key in self.measure_params:
                 if measure_output[0][i][key] is not None:
                     measure_dic[key] = [
                         measure_output[j][i][key] for j in range(len(measure_output))
                     ]
             measure_results[f.__name__] = measure_dic
+            if self.save_path is not None:
+                if not os.path.exists(os.path.join(self.save_path, f.__name__)):
+                    os.mkdir(os.path.join(self.save_path, f.__name__))
+
+                for key in ["segmentation", "deblended_images"]:
+                    if key in measure_dic.keys():
+                        np.save(os.path.join(self.save_path, f.__name__, key), measure_dic[key])
```

Comment:
Thanks for introducing the `os.path.join` which I believe should be equivalent to using `pathlib`.

Reply:
indeed, it's the old way, but for this case it works well